lazy_dataset/core.py

```diff
@@ -495,11 +588,11 @@ def zip(self, *others) -> 'ZipDataset':
             >>> ds4  # doctest: +ELLIPSIS
                     DictDataset(len=2)
                     DictDataset(len=2)
-                  ZipDataset()
+                  KeyZipDataset()
```

Comment:
Should this be `ZipDataset`?

Reply:
in that case the return of items() would not be indexable by keys

Reply:
Ah, sorry. The context that Github showed was confusing. Your change is correct.